lib/linguist/languages.yml

```
@@ -2701,6 +2701,14 @@ Jolie:
   ace_mode: text
   tm_scope: source.jolie
   language_id: 998078858
+Jq:
```

Shouldn't this be

```suggestion
JQ:
```

instead?

---

Can do, what is the convention? i think jq commonly is all lowercase but it seems the yml file uses captial keys?

---

If a language's name uses non-standard or unusual casing (as [jq](https://stedolan.github.io/jq/) does), then we use that; otherwise, **<ins>L</ins>abel <ins>C</ins>ase** is used (unless it's an obvious acronym, like HTML).

Aside from renaming the YAML entry, the `samples/Jq` directory will need to be renamed to use the updated capitalisation. Now, committing this might be tricky if you're using an OS with a case-insensitive filesystem (e.g., macOS). Running the following commands from the root of your fork should do the trick:

~~~sh
git config --local --add core.ignorecase false
git add --all samples/Jq samples/JQ
~~~

---

Bloody hell, I gave you the wrong capitalisation. 😰 That last line should have been

~~~sh
git mv samples/Jq samples/jq
~~~

So you'll now have to run

~~~sh
git mv samples/JQ samples/jq
~~~

Apologies for the screw-up. 😞

---

No worries, should be "jq" as yml key also?

---

Sorry, I missed that last question.

Yes. Everywhere the language's name appears should match the exact characters used in `languages.yml`: Heuristics, tests, the name of the `samples/` subdirectory, the grammar-index, etc.

The only exception to this is the `F*` language, which necessitated a [new field](https://github.com/github/linguist/blob/76c397c7ad07319e766fb1445b8a2307af2f586b/lib/linguist/languages.yml#L3-L5) that allowed us to use filename-safe characters for [`./samples/Fstar/](https://github.com/github/linguist/blob/76c397c7ad07319e766fb1445b8a2307af2f586b/lib/linguist/languages.yml#L1502). However, this is an edge-case, and one you'll almost certainly never have to worry or care about.

---

Ok, i assumed so, so already changed it